components/TransactionForm.js

```diff
@@ -0,0 +1,346 @@
+  grid-area: nameInput;
+  padding: 4px 12px;
+  border: 1px solid var(--dark-grey-color);
+  border-radius: 24px;
+  background-color: white;
+  color: #141414;
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+  transition: all 0.2s ease-in-out;
+
+  &:hover {
+    background-color: var(--light-bg-color);
+  }
+
+  &:focus {
+    border-color: var(--accent-color);
+    outline: none;
+    box-shadow: 0 0 4px rgba(70, 134, 205, 0.8);
+  }
+`;
+
+const StyledAmountLabel = styled.label`
+  grid-area: amountLabel;
+`;
+
+const StyledCurrencyInput = styled(CurrencyInput)`
+  grid-area: amountInput;
+  padding: 4px 12px;
+  border: 1px solid var(--dark-grey-color);
+  border-radius: 24px;
+  background-color: white;
+  color: var(--text-color-dark);
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+  transition: all 0.2s ease-in-out;
+
+  &:hover {
+    background-color: var(--light-bg-color);
+  }
+
+  &:focus {
+    border-color: var(--accent-color);
+    outline: none;
+    box-shadow: 0 0 4px rgba(70, 134, 205, 0.8);
+  }
+`;
+
+const StyledCategoryLabel = styled.label`
+  grid-area: categoryLabel;
+`;
+
+const StyledCategorySelect = styled.select`
+  grid-area: categoryInput;
+  padding: 4px 12px;
+  border: 1px solid var(--dark-grey-color);
+  border-radius: 24px;
+  background-color: white;
+  color: var(--text-color-dark);
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+  transition: all 0.2s ease-in-out;
+
+  &:hover {
+    background-color: var(--light-bg-color);
+  }
+
+  &:focus {
+    border-color: var(--accent-color);
+    outline: none;
+    box-shadow: 0 0 4px rgba(70, 134, 205, 0.8);
+  }
+`;
+
+const StyledTypeLabel = styled.label`
+  grid-area: typeLabel;
+`;
+
+const StyledRadioInput = styled.input`
+  grid-area: typeInput;
+`;
+
+const StyledRadioLabel = styled.label`
+  grid-area: radioLabel;
+`;
+
+const ErrorMessage = styled.p`
+  grid-area: typeErrorMessage;
+  color: red;
+  font-size: 10px;
+  margin-bottom: 4px;
+`;
+
+const StyledDateLabel = styled.label`
+  grid-area: dateLabel;
+`;
+
+const StyledDateInput = styled.input`
+  grid-area: dateInput;
+  padding: 4px 12px;
+  border: 1px solid var(--dark-grey-color);
+  border-radius: 24px;
+  background-color: white;
+  color: var(--text-color-dark);
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+  transition: all 0.2s ease-in-out;
+
+  &:hover {
+    background-color: var(--light-bg-color);
+  }
+
+  &:focus {
+    border-color: var(--accent-color);
+    outline: none;
+    box-shadow: 0 0 4px rgba(70, 134, 205, 0.8);
+  }
+`;
+
+const StyledLegend = styled.legend`
+  padding: 6px;
+  text-align: center;
+  font-weight: 500;
+`;
+
+const StyledToggleButton = styled.div`
+  display: flex;
+  justify-content: space-around;
+  align-items: center;
+  font-size: 16x;
+  height: 24px;
+  background-color: var(--light-bg-color);
+  border: 1px solid var(--dark-grey-color);
+  border-radius: 30px;
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+
+  input {
+    display: none;
```

Comment:
An error message appeared in the console (see screenshot). I suspect this is because the `input` fields are not displayed. You could perhaps get around the problem with `opacity: 0`? I'm not exactly a CSS expert, but you can give it a try. :wink:

![Bildschirmfoto 2024-10-07 um 14 00 50](https://github.com/user-attachments/assets/798d2b9f-b875-4f14-b84e-4d75b16d6b33)